clients/src/main/java/org/apache/kafka/common/requests/JoinGroupResponse.java

```diff
@@ -65,7 +66,17 @@ public JoinGroupResponse(short errorCode,
                              String memberId,
                              String leaderId,
                              Map<String, ByteBuffer> groupMembers) {
-        super(new Struct(CURRENT_SCHEMA));
+        this(CURRENT_VERSION, errorCode, generationId, groupProtocol, memberId, leaderId, groupMembers);
```

Comment:
Requests aren't considered public API, right? If not, then a different convention seems good to be more defensive.

But it probably makes sense to keep in a different PR, this is already quite extensive... If we actually follow up on this, there's no immediate risk with this current version.